pages/_app.js

```diff
@@ -69,6 +69,11 @@ export default function App({ Component, pageProps }) {
 
   const [showPlantFilterSection, setShowPlantFilterSection] = useState(false);
 
+  const [searchQuery, setSearchQuery] = useState("");
+  const [searchPage, setSearchPage] = useState(""); 
+  const [searchResults, setSearchResults] = useState([]);
+  const [noSearchResults, setNoSearchResults] = useState(false);
```

Comment:
I want you to think about your states and which if them are really necessary. React provides a very handy "checklist" to find out if a state is needed or not. Since every state change causes a re-render it's inevitable to keep an eye on the performance. To check if a state is needed or not, you can follow this checklist:

1. Does it remain unchanged over time? If so, it isn’t state.
2. Is it passed in from a parent via props? If so, it isn’t state.
3. Can you compute it based on existing state or props in your component? If so, it definitely isn’t state!

Let's apply this to your states:

1. searchQuery:
- Changes over time, is not passed as a prop and can't be computed - it's a state! ✅
- In your case: the search string

2. searchPage
- Does not change and can be passed as a prop/computed - it's not a state ❌
- In your case: can be derived from the router

3. searchResults
- Changes over time, but can be passed as a prop/computed - it's not a state ❌
- In your case: can be computed based on the original plant list

4. noSearchResults
- Changes over time, but can be passed as a prop/computed - it's not a state ❌
- In your case: can be computed based on the original plant list

If you want to know more about it just check out this chapter in the React documentation: [Find the minimal but complete representation of UI state](https://react.dev/learn/thinking-in-react#step-3-find-the-minimal-but-complete-representation-of-ui-state)